examples/cpp/fill-grid-pp2hadron.cpp

```diff
@@ -0,0 +1,245 @@
+/// Example used in the PineAPPL v1 paper ///
+#include <cstdint>
+#include <pineappl_capi.h>
+
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <string>
+#include <vector>
+
+struct Psp2to2Hadron {
+    double s;
+    double t;
+    double u;
+    double x1;
+    double x2;
+    double z;
+    double pt_hadron;
+    double y_hadron;
+    double jacobian;
+};
+
+double me_gg2qqbar(double s, double t, double u) {
+    (void) s; // ignore dummy variable
+    double as2 = 0.118 * 0.118;
+    double PI2 = M_PI * M_PI;
```

Comment:
Do not use `M_PI`, as it's not part of any C++ standard. You can use `std::acos(-1.0)`, which GCC evaluates at compile time.